client/remote_test.go

```diff
@@ -104,55 +105,105 @@ func TestMain(t *testing.T) {
 
 	// register both public keys with empty remote server so
 	// DefaultRemote will be used
-	if pemBytes, err = ioutil.ReadFile(rsaPubKey); err != nil {
-		t.Fatal(err)
-	}
-	p, _ = pem.Decode(pemBytes)
-	if pub, err = x509.ParsePKIXPublicKey(p.Bytes); err != nil {
+	registerCertFile(rsaPubKey, t)
+
+	registerCertFile(ecdsaPubKey, t)
+
+}
+
+func TestRemoteGroup(t *testing.T) {
+	_, err := c.Dial(rsaSKI)
+	if err != nil {
 		t.Fatal(err)
 	}
-	if rsaSKI, err = gokeyless.GetSKI(pub); err != nil {
+
+	_, err = c.Dial(ecdsaSKI)
+	if err != nil {
 		t.Fatal(err)
 	}
-	if rsaKey, err = c.RegisterPublicKey("", pub); err != nil {
-		t.Fatal(err)
+}
+
+func TestBadRemote(t *testing.T) {
+	// clear cached remotes and set a bad remote for the client
+	c.DefaultRemote = deadRemote
+	c.servers = map[string]Remote{}
+	c.remotes = map[gokeyless.SKI]Remote{}
+
+	// register the ECDDSA certificate again with the default broken remote
+	registerCertFile(ecdsaPubKey, t)
+
+	_, err := c.Dial(ecdsaSKI)
+	if err == nil {
+		t.Fatal("bad remote management")
 	}
+}
 
-	if pemBytes, err = ioutil.ReadFile(ecdsaPubKey); err != nil {
+func TestSlowServer(t *testing.T) {
+	// Setup keyless server
+	s2, err := server.NewServerFromFile(serverCert, serverKey, keylessCA,
```

Comment:
Could we bind to `0` instead of binding to a static port? Avoids potential conflicts when testing in a shared env.